hatchways/team-pine
# Create reviews front end and integrate with back end

### What this PR does:
- Reviews are now retrieved from the database and displayed on profile details page. The first three are displayed in the right grid section, if there are more a button will appear that shows a modal with all reviews.
- Review form was added below the review display section. Rating (default 3) is required but text is optional. It is fully integrated to create a review on the back end.
- Ratings are calculated when retrieving reviews.

### Screenshots / Videos:
![image](https://user-images.githubusercontent.com/34494229/152681529-22f9753a-9338-47d3-961b-4672be88e2da.png)
![image](https://user-images.githubusercontent.com/34494229/152681539-4e6bc9bc-c555-47b5-b170-ee43ae5afea0.png)
![image](https://user-images.githubusercontent.com/34494229/152681577-cb725756-1f45-44e9-a759-6e93374e055c.png)
![image](https://user-images.githubusercontent.com/34494229/152681599-6405cfc1-2551-4adc-9adc-eeff4ead8fc9.png)

### Any information needed to test this feature:
- You must have at least 1 sitter and 4 non sitter accounts to see all parts of this feature. You will need to add ratings to the sitter account if it was created before this PR. Default is 2.5 in the Profile model.
- Navigate to the profile details page of the sitter account (either directly with /profile/{profileId} or by searching on the listings page).
- Create a review on the review form. You should see the review appear on the page. Comments are optional.
- Repeat the previous two steps on at least 3 different accounts to see the modal button appear (you can't create multiple on the same profile with the same account).
- Try to create two reviews on the same account to see validation error.

closes #15
